### ISSUE TITLE
[Bug]: Linux `defaultPath` does not work on file dialogs for opening

### ISSUE THREAD

**Issue by bpasero** [electron/electron#43310]
### Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

30.x

### What operating system(s) are you using?

Ubuntu

### Operating System Version

Ubuntu 22.04

### What arch are you using?

arm64 (including Apple Silicon)

### Expected Behavior

A `defaultPath` passed to a open dialog should open in that folder.

### Actual Behavior

Dialogs always open here:

![Image](https://github.com/user-attachments/assets/f28cac84-ac77-42a5-af75-009e1252d01e)

### Additional Information

This issue is specific both to the open file dialog when using the dialog portal implementation.

Repro:

```js
const { app, BrowserWindow, dialog } = require('electron/main')

app.whenReady().then(async () => {
  const mainWindow = new BrowserWindow({ height: 600, width: 600 })

  // Show an "Open File" dialog and attempt to open
  // the chosen file in our window.
  try {
    const { filePaths, canceled } = await dialog.showOpenDialog(mainWindow, {
      properties: ['openFile'],
      defaultPath: '/home/parallels/GitDevelopment/vscode'
    })
    if (canceled) {
      console.log('Dialog was canceled')
    } else {
      const file = filePaths[0]
      mainWindow.loadFile(file)
    }
  } catch (err) {
    console.log(err)
  }
})
```

**Comment by codebytere** [comment]
@bpasero could you try updating your version of xdg-desktop-portal and xdg-desktop-portal-gtk? that would confirm my theory that it's to do with upstream support added recently!

**Comment by bpasero** [comment]
Any steps?

**Comment by sebastianvitterso** [comment]
I'm deploying an electron application (internal tool) to a RHEL8-system with 
```
$ /usr/libexec/xdg-desktop-portal --version
xdg-desktop-portal 1.12.6
```

and updating seems to be an enterprise management nightmare. 

Will this bug persist, then? 

We have two versions of our application deployed, one on Electron 28, and the newer one on Electron 30, and the bug only exists in Electron 30, so something changed between those two versions. Can anything be done to mitigate the bug on the Electron 30 version?

**Comment by sebastianvitterso** [comment]
Is there a cli-switch when packaging, or would I have to use that when launching the built application? Any documentation on the switch itself? I'm not entirely sure what to google.

**Comment by ckerr** [comment]
Just to check in with another datapoint, it seems to be working correctly for me on Ubuntu 24.04 x64.

This is possibly specific either to 22.04 or the version or GTK and/or xdg-desktop-portal that 22.04 ships with?

**Comment by codebytere** [comment]
@ckerr see https://github.com/electron/electron/issues/43310#issuecomment-2286200814 - it's to do with the xdg-desktop-portal version!

**Comment by riccardomanfrin** [comment]
FWIW: I'm having the issue on every version of vscode later than `1.89.1-1715060508`  (which is based on Electron `28.2.8`)

My version of `xdg-desktop-portal`:

```
$>  /usr/libexec/xdg-desktop-portal --version
xdg-desktop-portal 1.14.4
```

I'd also like to know if there's anything to work around it.

**Comment by sebastianvitterso** [comment]
@codebytere 
> You can provide a cli switch to use the gtk dialog path, which should be precisely the same functionality as before.

Hey, I'm not entirely sure what this means - is this something I can provide the compiler, or what do I have to do?

**Comment by codebytere** [comment]
@sebastianvitterso I've figured out a better approach that shouldn't require you to do anything - PR up soon!

**Comment by sebastianvitterso** [comment]
Thanks for resolving this @codebytere ! 🥳

I couldn't find anything specific relating to how often Electron releases patch versions, but when do you think this will be released? We're on the 32.x.y channel at this time, so I guess that just means when will 32.0.3 be released?

**Comment by lingo** [comment]
I have this problem too.

- Pop OS v22.04 LTS 
- Electron v42.3.3, 
- xdg-desktop-portal-gtk v1.14.0

Initially the dialog always opened in 'Recent' mode, but I changed the gnome settings to use cwd.

```
gsettings set org.gtk.Settings.FileChooser startup-mode 'cwd'
```

However, in both cases, the `defaultPath` option seems ignored. 

I have tried to disable use of XDG portal as [documented](https://www.electronjs.org/docs/latest/api/dialog#dialogshowopendialogwindow-options) :

> On Linux defaultPath is not supported when using portal file chooser dialogs unless the portal backend is version 4 or higher. You can use --xdg-portal-required-version [command-line switch](https://www.electronjs.org/docs/latest/api/command-line-switches#--xdg-portal-required-versionversion) to force gtk or kde dialogs.

Running both with, and without `--xdg-required-portal-version=4`, gives the following warning in logs:

> Portal availability checked before test was complete

stacktrace indicates it comes from this line https://github.com/electron/electron/blob/v42.3.3/shell/browser/ui/file_dialog_linux_portal.cc#L124




Reproduce issue: https://gist.github.com/f3f24a58330282d166fb41e6475f201b
